src/query/models/tag.go

```diff
@@ -236,54 +242,24 @@ func (t Tags) IDWithKeys(includeKeys ...string) uint64 {
 }
 
 // TagsWithKeys returns only the tags which have the given keys
-func (t Tags) TagsWithKeys(includeKeys []string) Tags {
+func (t Tags) TagsWithKeys(includeKeys [][]byte) Tags {
 	return t.tagSubset(includeKeys, true)
 }
 
 // WithoutName copies the tags excluding the name tag
 func (t Tags) WithoutName() Tags {
-	return t.TagsWithoutKeys([]string{MetricName})
+	return t.TagsWithoutKeys([][]byte{MetricName})
 }
 
 // Get returns the value for the tag with the given name.
-func (t Tags) Get(key string) (string, bool) {
+func (t Tags) Get(key []byte) ([]byte, bool) {
 	for _, tag := range t {
-		if tag.Name == key {
+		if bytes.Equal(tag.Name, key) {
 			return tag.Value, true
 		}
 	}
 
-	return "", false
-}
-
-// FromMap returns new sorted tags from the given map.
-func FromMap(m map[string]string) Tags {
-	l := make(Tags, 0, len(m))
-	for k, v := range m {
-		l = append(l, Tag{Name: k, Value: v})
-	}
-
-	return Normalize(l)
-}
-
-// TagMap returns a tag map of the tags.
-func (t Tags) TagMap() map[string]Tag {
-	m := make(map[string]Tag, len(t))
-	for _, tag := range t {
-		m[tag.Name] = tag
-	}
-
-	return m
-}
-
-// StringMap returns a string map of the tags.
-func (t Tags) StringMap() map[string]string {
-	m := make(map[string]string, len(t))
-	for _, tag := range t {
-		m[tag.Name] = tag.Value
-	}
-
-	return m
+	return []byte{}, false
```

Comment:
`return nil, false` is slightly more optimal (no allocs)

Reply:
Good call